teps/0115-tekton-catalog-git-based-versioning.md

```diff
@@ -0,0 +1,398 @@
+- Bumping versions of resources is cumbersome which encourages making "minor" changes in existing versions. Some changes
+that seem minor cause surprising breaking changes for users, such as described in [tektoncd/catalog#784][784].
+- Resources that are frequently updated with new versions end up with a lot of duplications because any change requires
+copying over everything into a new directory. For example, [git-clone][git-clone] `Task` has seven versions with a lot
+of duplication across versions.
+
+Previously, there was tight coupling between the organization and how users referenced resources which made it hard to 
+change the organization of the Catalog without breaking users. In [TEP-0110][tep-0110], we decoupled the Catalog 
+organization from resource reference. Now, we can create a new organization and versioning contract for Catalogs to
+address the challenges with the directory-based organization and versioning.
+
+The aim of this TEP is to create a new organization and versioning for Tekton Catalogs, building on the prior work and 
+experiences from [TEP-0003][tep-0003]. In [TEP-0079: Tekton Catalog Support Tiers][tep-0079], we plan to add a Tekton 
+Official Catalog which will use this new organization and versioning contract - therefore, this proposal is a blocker 
+for [TEP-0079][tep-0079].
+
+## Proposal
+
+We propose supporting a flat organization and git-based versioning of Tekton Catalogs.
+
+### Flat Organization
+
+In this organization contract, we'll use a flat structure instead of the directory structure. The example shown above
+with the two `Tasks` (argocd and golang-build) and one `Pipeline` (go-release) would be organized as such:
+
+```shell
+./task/
+  /argocd
+    /README.md
+    /argocd.yaml
+    /samples
+    /tests
+    /OWNERS
+  /golang-build
+    /README.md
+    /golang-build.yaml
+    /samples
+    /tests
+    /OWNERS
+./pipeline/
+  /go-release
+    /README.md
+    /go-release.yaml
+    /samples
+    /tests
+    /OWNERS
+```
+
+Note that there are no changes to the contents of the resources themselves, only the organization of Catalogs.
+
+### Git-Based Versioning
+
+#### Versioning via Tags
+
+Versioning of resources will be done through creating tags in the Catalogs. The versions in the Tekton Hub should
+continue to use the simple version of [semver][semver]: `{major}.{minor}`. We propose that the tags use the semantic
+versioning model - `v<major-version>.<minor-version>`. Minor versions are made to signal backwards-compatible changes,
+while major versions are made to signal backwards-incompatible changes. As such, the tags in the catalogs will map to
+versions in the Hub e.g. tag "v0.7" in a Catalog maps to version "0.7" in the Hub.
+
+#### One Version per Catalog
+
+Note that version tags apply to the whole repository. Therefore, we propose that Catalogs should contain one or
+multiple resource that have the same versioning. Instead of the centralized Catalog from the community, we will
```

Comment:
I think it may be worth expanding on the "same versioning" idea with some examples.
I can imagine very common tasks like `git-clone` living in their own repository, maintained by the community.
In large organisations, the platform team could maintain a catalog of tasks and release it on a regular base to the dev teams. 
Having one repo per task there would imply a significant administrative overhead and ultimately discourage sharing and re-use of tasks, so I think it would be useful to still consider the case of multiple tasks and pipelines versioned together with a versioning strategy that is not bound to the single resource.

Reply:
fleshed out the Buildpacks and eBay examples, and added an Openshift example

the Buildpacks and Openshift examples demonstrate the case of multiple tasks and pipelines versioned together